src/main/java/org/jabref/logic/specialfields/SpecialFieldsUtils.java

```diff
@@ -46,9 +46,12 @@
 
         Optional<Keyword> newValue = entry.getField(specialField.getFieldName()).map(Keyword::new);
         KeywordList keyWords = specialField.getKeyWords();
-
-        Optional<FieldChange> change = entry.replaceKeywords(keyWords, newValue, keywordDelimiter);
-        change.ifPresent(changeValue -> fieldChanges.add(changeValue));
+        if (newValue.isPresent()) {
```

Comment:
You can chain nested optionals with map, it will then only be executed if a value is present